src/main/java/dk/kb/util/webservice/Service2ServiceRequest.java

```diff
@@ -44,24 +44,30 @@ public class Service2ServiceRequest {
     * @param uri the full URI with path and parameters set.
     * @param httpMethod The http-method to use for the service call. GET, POST, DELETE etc.
     * @param objectClass The DTO type that the response should be parsed to.
-    * @return DtoObject (objectClass) of the same type at given as input. 
+    * @param postJsonDto If not the DTO will be send as POST. Set to null if API method does not require a DTO. 
+    * @return DtoObject (objectClass) of the same type at given as input. Use null as input for void API methods that has no return dto.
     * @throws ServiceException If anything unexpected happens.   
     **/    
-    public static <T> T httpCallWithOAuthToken (URI uri , String httpMethod, T objectClass) throws ServiceException {                 
+    public static <T> T httpCallWithOAuthToken (URI uri , String httpMethod, T objectClass, Object postJsonDto) throws ServiceException {                 
         //The token (message) will be set if the service method that initiated this call required OAuth token. 
-        String token= (String) JAXRSUtils.getCurrentMessage().get(OAuthConstants.ACCESS_TOKEN_STRING); 
+        String token= getOAuth2Token();
         Map<String, String> requestHeaders= new HashMap<String, String>();
+            if (postJsonDto != null) { 
+                requestHeaders.put("Content-Type", "application/json");
+                                
+            }   
```

Comment:
Maybe add an inline comment on why this is only important if != null

Reply:
added